server/routerlicious/packages/routerlicious-base/src/nexus/runner.ts

```diff
@@ -0,0 +1,222 @@
+import { LumberEventName, Lumberjack, LogLevel } from "@fluidframework/server-services-telemetry";
+import {
+	configureWebSocketServices,
+	ICollaborationSessionEvents,
+} from "@fluidframework/server-lambdas";
+import { runnerHttpServerStop } from "@fluidframework/server-services-shared";
+import * as app from "./app";
+
+export class NexusRunner implements IRunner {
+	private server: IWebServer;
+	private runningDeferred: Deferred<void>;
+	private stopped: boolean = false;
+	private readonly runnerMetric = Lumberjack.newLumberMetric(LumberEventName.NexusRunner);
+
+	constructor(
+		private readonly serverFactory: IWebServerFactory,
+		private readonly config: Provider,
+		private readonly port: string | number,
+		private readonly orderManager: IOrdererManager,
+		private readonly tenantManager: ITenantManager,
+		private readonly socketConnectTenantThrottler: IThrottler,
+		private readonly socketConnectClusterThrottler: IThrottler,
+		private readonly socketSubmitOpThrottler: IThrottler,
+		private readonly socketSubmitSignalThrottler: IThrottler,
+		private readonly storage: IDocumentStorage,
+		private readonly clientManager: IClientManager,
+		private readonly metricClientConfig: any,
+		private readonly throttleAndUsageStorageManager?: IThrottleAndUsageStorageManager,
+		private readonly verifyMaxMessageSize?: boolean,
+		private readonly redisCache?: ICache,
+		private readonly socketTracker?: IWebSocketTracker,
+		private readonly tokenRevocationManager?: ITokenRevocationManager,
+		private readonly revokedTokenChecker?: IRevokedTokenChecker,
+		private readonly collaborationSessionEventEmitter?: TypedEventEmitter<ICollaborationSessionEvents>,
+	) {}
+
+	// eslint-disable-next-line @typescript-eslint/promise-function-async
+	public start(): Promise<void> {
+		this.runningDeferred = new Deferred<void>();
+
+		// Create the HTTP server and attach nexus to it
+		const nexus = app.create(this.config, this.collaborationSessionEventEmitter);
+		nexus.set("port", this.port);
+
+		this.server = this.serverFactory.create(nexus);
+
+		const httpServer = this.server.httpServer;
+
+		const maxNumberOfClientsPerDocument = this.config.get(
+			"nexus:maxNumberOfClientsPerDocument",
+		);
+		const numberOfMessagesPerTrace = this.config.get("nexus:numberOfMessagesPerTrace");
+		const maxTokenLifetimeSec = this.config.get("auth:maxTokenLifetimeSec");
+		const isTokenExpiryEnabled = this.config.get("auth:enableTokenExpiration");
+		const isClientConnectivityCountingEnabled = this.config.get(
+			"usage:clientConnectivityCountingEnabled",
+		);
+		const isSignalUsageCountingEnabled = this.config.get("usage:signalUsageCountingEnabled");
+
+		httpServer.on("error", (error) => this.onError(error));
+		httpServer.on("listening", () => this.onListening());
+		httpServer.on("upgrade", (req, socket, initialMsgBuffer) => {
+			const metric = Lumberjack.newLumberMetric("WebSocket Connections", {
```

Comment:
for metric name, the convention we follow is to use camelCase and no space.
So, maybe let's use WebsocketConnectionCount ?

Reply:
Also, it looks like we're using `WebSocketConnections` metric name 3 times with a new metric for each. Would it be better to create 1 `WebSocketConnection` metric, then `success`/`fail` on `close`/`error` respectively? Otherwise we are missing out on helpful Metric telemetry like duration, and also creating an unnecessary amount of different metrics.